src/drivers/interrupt/bpi_intc.h

```diff
@@ -0,0 +1,14 @@
+/**
+ * @file
+ * @brief Banana Pi interrupt controller HAL definitions.
+ *
+ * @date 15.04.15
+ * @author Dmitrii Petukhov
+ */
+
+#ifndef IRQCTRL_BPI_INTC_IMPL_H_
+#define IRQCTRL_BPI_INTC_IMPL_H_
+
+#define __IRQCTRL_IRQS_TOTAL 123
```

Comment:
Yes, there are. :)
Check this out: http://lib.chipdip.ru/974/DOC000974439.pdf (page 147)
I've answered the second question above.